src/main/java/com/ldtteam/structurize/client/gui/WindowSwitchPack.java

```diff
@@ -202,26 +245,43 @@ public void updatePacks()
             @Override
             public int getElementCount()
             {
-                return (int) Math.ceil(filteredPackMetas.size() / 2.0);
+                return sortedPacks.size() / 2;
             }
 
             @Override
             public void updateElement(final int index, final Pane rowPane)
             {
                 int metaStart = index * 2;
 
-                final StructurePackMeta packMeta = filteredPackMetas.get(metaStart);
-                fillForMeta(rowPane, packMeta, "1");
+                final Object obj1 = sortedPacks.get(metaStart);
+                final Object obj2 = sortedPacks.get(metaStart + 1);
 
-                if (filteredPackMetas.size() > metaStart + 1)
+                rowPane.findPaneByID("box0").hide();
+                if (obj1 instanceof StructurePackMeta)
+                {
+                    fillForMeta(rowPane, (StructurePackMeta) obj1, "1");
+                    rowPane.findPaneByID("box1").show();
+                }
+                else
                 {
-                    fillForMeta(rowPane, filteredPackMetas.get(metaStart + 1), "2");
+                    rowPane.findPaneByID("box1").hide();
+                }
+
+                if (obj2 instanceof StructurePackMeta)
+                {
+                    fillForMeta(rowPane, (StructurePackMeta) obj2, "2");
                     rowPane.findPaneByID("box2").show();
                 }
                 else
                 {
                     rowPane.findPaneByID("box2").hide();
                 }
+
+                if (obj1 instanceof String)
+                {
+                    rowPane.findPaneByID("box0").show();
+                    rowPane.findPaneOfTypeByID("category", Text.class).setText(Component.literal(((String) obj1).substring(0, 1).toUpperCase(Locale.US) + ((String) obj1).substring(1)));
```

Comment:
This component is sus - many substrings - ideally fnc for capitalize, also instanceof pattern for string.. and above too